C/include/c4ReplicatorTypes.h

```diff
@@ -184,10 +188,32 @@ C4API_BEGIN_DECLS
 #endif // COUCHBASE_ENTERPRISE
 
 
+    typedef struct C4ReplicationCollection {
+        C4CollectionSpec collection;
+
+        C4ReplicatorMode                    push;              ///< Push mode (from db to remote/other db)
+        C4ReplicatorMode                    pull;              ///< Pull mode (from db to remote/other db).
+
+        // Following options should be encoded into the optionsDictFleed per-collection
+        //#define kC4ReplicatorOptionDocIDs           "docIDs"   ///< Docs to replicate (string[])
+        //#define kC4ReplicatorOptionChannels         "channels" ///< SG channel names (string[])
+        //#define kC4ReplicatorOptionFilter           "filter"   ///< Pull filter name (string)
+        //#define kC4ReplicatorOptionFilterParams     "filterParams"  ///< Pull filter params (Dict[string])
+        //#define kC4ReplicatorOptionSkipDeleted      "skipDeleted" ///< Don't push/pull tombstones (bool)
+        //#define kC4ReplicatorOptionNoIncomingConflicts "noIncomingConflicts" ///< Reject incoming conflicts (bool)
+        //#define kC4ReplicatorCheckpointInterval     "checkpointInterval" ///< How often to checkpoint, in seconds (number)
+        //
+        C4Slice                             optionsDictFleece;
```

Comment:
I don't know if those beside docIDs, Channels, Filter/FilterParams should be configured per collection. Can you check with Jens?

Reply:
It doesn't seem useful to have the others be per-collection, as far as I can tell.

Reply:
I'll remove others from the above comment.